src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,22 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    add: (state, action) => {
+      state.push(action.payload);
+    },
+    remove: (state, action) => {
+      const newstate = state.filter((item) => item.id !== action.payload);
+      const updateIndex = newstate.map((item, index) => ({ ...item, id: index + 1 }));
+      return updateIndex;
+    },
+  },
```

Comment:
- Great job for adding and removing the state, I think you have to push and filter from the `book` state instead of the global state.
`state.book.push(action.payload);` and `const newstate = state.book.filter((item) => item.id !== action.payload);`
- [Optional] Kindly, I think it would be better to use descriptive names for your reducers like `addbook` and `removeBook` to make it easy to access 👍